### PR TITLE
Remove unnecessary usage of immutableset

### LINKED ISSUE #906: Unnecessary use of immutableset in GraphMatchingState.__init__ takes 2-3% of time
[In __init__ of `GraphMatchingState`](https://github.com/isi-vista/adam/blob/f7f4c5bb5d3ebff790a21cc8b812de4c5883da8a/adam/perception/_matcher.py#L897) (which is created repeatedly, once for each search step), we have the following pattern repeated three times:

```python
new_nodes_0: ImmutableSet[Any] = immutableset(
                flatten(
                    [
                        predecessor
                        for predecessor in GM.graph.predecessors(node)
                        if predecessor not in GM.graph_node_to_pattern_node
                    ]
                    for node in GM.graph_node_to_pattern_node
                )
            )
            for node in new_nodes_0:
                if node not in GM.graph_nodes_in_or_preceding_match:
                    GM.graph_nodes_in_or_preceding_match[node] = self.depth
```

We expect the collisions that `immutableset` is trying to deduplciated to be (a) relatively rare and (b) not to matter anyway since the assignment is idempotent.  Dropping the set and just using nested for loops should be faster, since creating an `immutableset` is relatively expensive.

### PR DESCRIPTION
**Pull request by spigo900**
Closes #906.

### CODE REVIEW

File: adam/perception/_matcher.py [hunk 1]
```diff
@@ -893,67 +893,40 @@ def __init__(self, GM: GraphMatching, graph_node=None, pattern_node=None) -> Non
...
-                )
-            )
-            for node in new_nodes_0:
-                if node not in GM.graph_nodes_in_or_preceding_match:
-                    GM.graph_nodes_in_or_preceding_match[node] = self.depth
+            for node in GM.graph_node_to_pattern_node:
+                for predecessor in GM.graph.predecessors(node):
+                    if (
+                        predecessor not in GM.graph_node_to_pattern_node
+                        and node not in GM.graph_nodes_in_or_preceding_match
```
Comment (gabbard) [inline adam/perception/_matcher.py:900]: Should this be:
```suggestion
                        and predecessor not in GM.graph_nodes_in_or_preceding_match
```

Comment (spigo900) [reply adam/perception/_matcher.py:900]: Hm, possibly, but isn't that what it was doing before?
[Line 908](https://github.com/isi-vista/adam/pull/907/files#diff-055afb1c07a0ca7f2989c5ab842afcefL908).

Comment (gabbard) [reply adam/perception/_matcher.py:900]: @spigo900 : The naming here is confusing. Note that `node` on line 908 is coming from `new_nodes_0` and therefore corresponds to `precedessor` (and not `node`) from the old list comprehension.

Comment (spigo900) [reply adam/perception/_matcher.py:900]: Ah, I see. Thanks!

Comment (spigo900) [reply adam/perception/_matcher.py]: @gabbard That fixed it.

**Pull request merged**